src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs

```diff
@@ -166,7 +166,7 @@ private void RemoveSuppressOperationForBlock(List<SuppressOperation> list, Synta
 
             for (int i = 0; i < list.Count; i++)
             {
-                if (list[i] != null && list[i].TextSpan.Start >= span.Start && list[i].TextSpan.End <= span.End)
+                if (list[i] != null && list[i].TextSpan.Start >= span.Start && list[i].TextSpan.End <= span.End && list[i].Option == SuppressOption.NoWrappingIfOnSingleLine)
```

Comment:
this is a pretty core method.  Can you add comments explaining why this behavior is correct?

Reply:
@CyrusNajmabadi ,

When setting these options:
```charp
changingOptions.Add(CSharpFormattingOptions.SpacesIgnoreAroundVariableDeclaration, true);
changingOptions.Add(CSharpFormattingOptions.WrappingKeepStatementsOnSingleLine, false);
```
and running this code:
```csharp
class Program
{
    public void FixMyType()
    {
        var    myint    =    0;
    }
}
```
This line of code is removing all three of these entries: 
![image](https://user-images.githubusercontent.com/25776963/56396628-961e6680-61f4-11e9-974d-83d550a912ac.png)

If I leave the NoSpacing entry in the list, it keeps the spacing intact as desired.

There might be another place that is more appropriate to fix it.  Any ideas where?

Reply:
The fix could also be this, `if (list[i] != null && list[i].TextSpan.Start >= span.Start && list[i].TextSpan.End <= span.End && list[i].Option != SuppressOption.NoSpacing)`. I'm not familiar enough with this code to know if other features are affected by the same setting.

Reply:
I would definitely ask @heejaechang about this.  I think he's the only one qualified to state the best place for this fix.

Reply:
I am not sure why code is written this way, but it looks like we are removing suppression operation based on option rather than add suppression operation based on option.

because it is blindly add all suppression rule first and then later try to remove those suppression added, we have this strange code flow.

I think we should have an issue opened saying we should do thing in opposite way where we add suppression rules based on option rather than remove rule based on options

Reply:
that being said, in current form, what @chborl did is right.

Reply:
>I think we should have an issue opened saying we should do thing in opposite way where we add suppression rules based on option rather than remove rule based on options

I filed this follow-up issue: #35377